.github/workflows/deploy.yaml

```diff
@@ -28,27 +31,162 @@ jobs:
+          echo "Fetched SCHEMA_VERSION: $SCHEMA_VERSION"
+        working-directory: '${{ matrix.value }}'
+      - name: Compare schema versions and determine new version
+        id: compare_versions
+        if: ${{ env.IS_MAIN }}
+        run: |
+          LAST_PUBLISHED_VERSION=${{ steps.get_last_tag.outputs.last_tag }}
+          SCHEMA_VERSION=${{ steps.get_schema_version.outputs.schema_version }}
+
+          echo "Fetched LAST_PUBLISHED_VERSION: $LAST_PUBLISHED_VERSION"
+          echo "Schema version: $SCHEMA_VERSION"
+
+          NUMERIC_LAST_PUBLISHED_VERSION=$(echo $LAST_PUBLISHED_VERSION | sed 's/.*-v//')
+          IFS='.' read -ra LAST_PUBLISHED_VERSION_PARTS <<< "$NUMERIC_LAST_PUBLISHED_VERSION"
+
+          MAJOR=${LAST_PUBLISHED_VERSION_PARTS[0]}
+          MINOR=${LAST_PUBLISHED_VERSION_PARTS[1]}
+          PATCH=${LAST_PUBLISHED_VERSION_PARTS[2]}
+
+          IFS='.' read -ra SCHEMA_VERSION_PARTS <<< "$SCHEMA_VERSION"
+          SCHEMA_MAJOR=${SCHEMA_VERSION_PARTS[0]}
+          SCHEMA_MINOR=${SCHEMA_VERSION_PARTS[1]}
+          SCHEMA_PATCH=${SCHEMA_VERSION_PARTS[2]}
+
+          # determine version increment
+          if [[ "$SCHEMA_MAJOR" -gt "$MAJOR" ]]; then
+            MAJOR=$((MAJOR + 1))
+            MINOR=0
+            PATCH=0
+            SHOULD_DEPLOY=true
+          elif [[ "$SCHEMA_MAJOR" -eq "$MAJOR" && "$SCHEMA_MINOR" -gt "$MINOR" ]]; then
+            MINOR=$((MINOR + 1))
+            PATCH=0
+            SHOULD_DEPLOY=true
+          elif [[ "$SCHEMA_MAJOR" -eq "$MAJOR" && "$SCHEMA_MINOR" -eq "$MINOR" && "$SCHEMA_PATCH" -gt "$PATCH" ]]; then
+            PATCH=$((PATCH + 1))
+            SHOULD_DEPLOY=true
+          else
+            echo "Schema version is not greater than last published version. No deployment needed."
+            SHOULD_DEPLOY=false
+          fi
+
+          NEW_PUBLISHED_VERSION="$MAJOR.$MINOR.$PATCH"
+          echo "NEW_PUBLISHED_VERSION=$NEW_PUBLISHED_VERSION" >> $GITHUB_OUTPUT
+          echo "should_deploy=$SHOULD_DEPLOY" >> $GITHUB_OUTPUT
+
+      - name: Verify new published version
+        run: |
+          echo "NEW_PUBLISHED_VERSION: ${{ steps.compare_versions.outputs.new_published_version }}"
+        working-directory: '${{ matrix.value }}'
+      - name: Set version in version.json
+        if: ${{ env.IS_MAIN }} && steps.compare_versions.outputs.should_deploy == 'true'
+        run: |
+          NEW_PUBLISHED_VERSION=${{ steps.compare_versions.outputs.new_published_version }}
+          SCHEMA_VERSION=${{ steps.get_schema_version.outputs.schema_version }}
+          echo "{\"schemaVersion\": \"$SCHEMA_VERSION\", \"publishedVersion\": \"$NEW_PUBLISHED_VERSION\"}" > version.json
+        working-directory: '${{ matrix.value }}'
+      - name: Set Version in version.ts for version entity
+        if: ${{ env.IS_MAIN }} && steps.compare_versions.outputs.should_deploy == 'true'
+        run: npm run set-version
```

Comment:
"why" comment:
```suggestion
        # Updating version.ts files because the subgraph handlers can't read .json directly
        run: npm run set-version
```